module/entities/helper.js

```diff
@@ -926,106 +927,86 @@ module.exports = class EntitiesHelper {
     ) {
         return new Promise(async (resolve, reject) => {
             try {
-
-                let queryObject = {};
-
-                queryObject["$match"] = {};
-
-                if (entityIds && entityIds.length > 0) {
-                    queryObject["$match"]["_id"] = {};
-                    queryObject["$match"]["_id"]["$in"] = entityIds;
-                }
-
-                if( aclData.length > 0 ) {
-                    queryObject["$match"]["metaInformation.tags"] = 
-                    { $in : aclData };
+                let entityDocuments = [];
+                let bodyData={
+                    "type" : entityType
+                };
+                
+                let entitiesData = await sunbirdService.learnerLocationSearch( bodyData );
+                
+                if( !entitiesData.data.response.length > 0 ) {
+                    return resolve(entitiesData.data.response) 
                 }
-
-                queryObject["$match"]["entityTypeId"] = entityTypeId;
-
-                queryObject["$match"]["$or"] = [
-                    { "metaInformation.name": new RegExp(searchText, 'i') },
-                    { "metaInformation.externalId": new RegExp("^" + searchText, 'm') },
-                    { "metaInformation.addressLine1": new RegExp(searchText, 'i') },
-                    { "metaInformation.addressLine2": new RegExp(searchText, 'i') }
-                ];
-
-                let entityDocuments = await database.models.entities.aggregate([
-                    queryObject,
-                    {
-                        $project: {
-                            name: "$metaInformation.name",
-                            externalId: "$metaInformation.externalId",
-                            addressLine1: "$metaInformation.addressLine1",
-                            addressLine2: "$metaInformation.addressLine2",
-                            districtName: "$metaInformation.districtName"
-                        }
-                    },
-                    {
-                        $facet: {
-                            "totalCount": [
-                                { "$count": "count" }
-                            ],
-                            "data": [
-                                { $skip: pageSize * (pageNo - 1) },
-                                { $limit: pageSize }
-                            ],
-                        }
-                    }, {
-                        $project: {
-                            "data": 1,
-                            "count": {
-                                $arrayElemAt: ["$totalCount.count", 0]
-                            }
+         
+                let immediateLocation = entitiesData.data.response;
```

Comment:
@VISHNUDAS-tunerlabs We need to give priority to entities word instead of the location. We are fetching the values from location search but still our key will be entity